### PR TITLE
Make the enclave interface and docs consistent

### PR DESCRIPTION
**Pull request by cmickeyb**
The json specifications for the enclave had gotten
out of date. This updates the documents and makes the
interface consistent with the documents. Updated field
names to be consistent with the parameters.

Also split up a few very long comments and docstrings.

Note that string lengths in the basetypes are incorrect for some of the types.

Signed-off-by: Mic Bowman <mic.bowman@intel.com>

### CODE REVIEW

File: eservice/docs/basetypes.json [hunk 1]
```diff
@@ -5,140 +5,101 @@
...
         },
+        "encoded-hash": {
+            "id": "#encoded-hash",
+            "description": [
+                "encoded sha256 hash"
+            ],
+            "type": "string",
+            "minLength": 22,
+            "maxLength": 22,
+            "format": "base64, urlsafe encoded 64 byte array"
```
Comment (prakashngit) [inline eservice/docs/basetypes.json]: what is minLength and maxLength here? isnt 256 bits ~ length 43 in base64?

File: python/pdo/contract/contract.py [hunk 2]
```diff
@@ -161,7 +163,7 @@ def create_update_request(self, request_originator_keys, expression, enclave_ser
             request_originator_keys,
             self,
             enclave_service=enclave_service,
-            expression = expression)
+            invocation_request = expression)
```
Comment (prakashngit) [inline python/pdo/contract/contract.py]: this change seems misleading, given that a request till now meant evaluate "expression on a specific state". Especially, with the new term invocation_response, which is likely interpreted as invocation_response = function(invocation_request) = function(expression, state)

File: eservice/docs/basetypes.json (hunk 1 above)
Comment (cmickeyb) [reply eservice/docs/basetypes.json]: i believe my comment was specifically that string lengths are not updated and out of scope for this PR.

File: python/pdo/contract/contract.py (hunk 2 above)
Comment (cmickeyb) [reply python/pdo/contract/contract.py]: i don't understand this comment. or why it is relevant?

File: eservice/docs/basetypes.json (hunk 1 above)
Comment (prakashngit) [reply eservice/docs/basetypes.json]: yes, missed that ! thank you

File: python/pdo/contract/contract.py (hunk 2 above)
Comment (prakashngit) [reply python/pdo/contract/contract.py]: was just referring to the change in the variable name from expression to invocation_request. i guess i am being a bit too formal on the choice of names. invocation_request is OK.

**Pull request merged**